opengrok-indexer/src/main/java/org/opengrok/indexer/web/LaunderUtil.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2020, Chris Fraire <cfraire@me.com>.
+ */
+
+package org.opengrok.indexer.web;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents a container for sanitizing methods for avoiding classifications as
+ * taint bugs.
+ */
+public class LaunderUtil {
+
+    /**
+     * Sanitize {@code value} where it will be used in subsequent OpenGrok
+     * (non-logging) processing.
+     * @return {@code null} if null or else {@code value} with "pattern-breaking
+     * characters" (tabs, CR, LF, FF) replaced as underscores (one for one)
+     */
+    public static String userInput(String value) {
+        if (value == null) {
+            return null;
+        }
+        return value.replaceAll("[\\n\\r\\t\\f]", "_");
```

Comment:
reuse the String literal ?

Reply:
also, userInput() and luceneQuery() can be refactored into using common underlying method ?

Reply:
Yes done